### ISSUE TITLE
🐛 [firebase_core_web] Method  not found: 'nopromiseToFuture'

### ISSUE THREAD

**Issue by ChaseGuru** [firebase/flutterfire#11577]
## Bug report

**Describe the bug**
Recently, my flutter projects using web are failing to build because of an error with a method from the js package not being found.  Using an older version of firebase fixes the issue.

### Steps to reproduce
Package with dependencies like:
```
  firebase_auth: ^4.6.3
  firebase_core: ^2.14.0
  firebase_crashlytics: ^3.3.3
```
```
flutter build web
```

### Expected behavior

NO error.

### Sample project

---

## Additional context

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

Pre upgrade:
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.10.6, on Ubuntu 22.04.2 LTS 6.2.0-26-generic, locale en_US.UTF-8)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[✓] Chrome - develop for the web
[✓] Linux toolchain - develop for Linux desktop
[✓] Android Studio (version 2022.1)
[✓] VS Code
[✓] Connected device (2 available)
[✓] Network resources

• No issues found!
```
Post upgrade:
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.13.3, on Ubuntu 22.04.2 LTS 6.2.0-26-generic, locale en_US.UTF-8)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[✓] Chrome - develop for the web
[✓] Linux toolchain - develop for Linux desktop
[✓] Android Studio (version 2022.1)
[✓] VS Code (version unknown)
    ✗ Unable to determine VS Code version.
[✓] Connected device (2 available)
[✓] Network resources

• No issues found!
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
PASTE OUTPUT INSIDE HERE
```

</details>

---

### Temporary fix:
I am able to compile with these dependencies:
```
  firebase_core: 2.15.0
  firebase_auth: 4.7.0
  cloud_firestore: 4.8.3
```

**Comment by ChaseGuru** [comment]
Basically, just documenting, for me, if you find the versions of the packages that correspond with 2.15.0 it works.

**Comment by ChaseGuru** [comment]
Fore crashlytics, I did 3.3.4.  But, it still resolved 2.7.0 of core_web for me, so I guess dependency override may be easier:
```
dependency_overrides:
  firebase_core_web: 2.6.0
```

**Comment by danagbemava-nc** [comment]
Hi @ChaseGuru, can you provide a complete minimal sample so that we can investigate this issue? I added the newest versions of the packages mentioned above to a new project but I was not able to reproduce the issue.
